src/java.base/share/classes/java/util/concurrent/locks/AbstractQueuedLongSynchronizer.java

```diff
@@ -277,6 +277,27 @@ private static void signalNextIfShared(Node h) {
         }
     }
 
+    /**
+     * Invokes acquire for as long as it throws exceptions, backing off using Unsafe.park
+     * @param node which to reacquire
+     * @param arg the acquire argument
+     */
+    private final void reacquire(Node node, long arg) {
```

Comment:
@AlanBateman @DougLea If we're concerned about SOE due to adding another stackframe by invoking reacquire we might consider ForceInline, but I was weary about doing so.